bot.py

```diff
@@ -712,6 +719,65 @@ async def clear_queue(self, message, _command_content):
         self._stop()
         await message.add_reaction(MusicBot.REACTION_EMOJI)
 
+    async def remove_song_from_deque(self, message, command_content):
+        """
+        Removes a specific song from the queue, based on its position therein
+        """
+        if len(self.media_deque) == 0:
+            await message.channel.send(":unamused: The queue is empty...")
+            return
+
+        arg = command_content.split(maxsplit=1)[0]
+
+        range_regex = re.compile(r"(\d+)-(\d+)")
+        range_regex_match = range_regex.match(arg)
+        if range_regex_match:
+            await self.remove_range_from_deque(message, range_regex_match)
+            return
+
+        if arg.lower() == "first":
+            song_index = 0
+        elif arg.lower() == "last":
+            song_index = -1
+        elif arg.isnumeric():
+            song_index = int(arg) - 1
+        else:
+            await message.channel.send(
+                f"{arg} isn't a valid song position/range :open_mouth:"
+            )
+            return
+
+        if song_index < -1 or song_index >= len(self.media_deque):
+            await message.channel.send(
+                "Pick something *in* the queue! :face_with_symbols_over_mouth:"
+            )
+            return
+
+        del self.media_deque[song_index]
+        await message.add_reaction(MusicBot.REACTION_EMOJI)
+
+    async def remove_range_from_deque(self, message, range_regex_match):
+        """
+        Receives a match regex and uses its 2 capture groups to remove a range of songs
+        from the queue.
+        """
+        range1 = int(range_regex_match.group(1))
+        range2 = int(range_regex_match.group(2))
+        range_start = min(range1, range2)
+        range_end = max(range1, range2)
+
+        if range_start < 1 or range_end > len(self.media_deque):
+            await message.channel.send(
+                f"{range_regex_match.group(0)} isn't a valid range :open_mouth:"
+            )
+            return
+
+        n_songs_to_remove = range_end - range_start + 1
+        for _ in range(n_songs_to_remove):
+            del self.media_deque[range_start - 1]
```

Comment:
Could you add a comment here why +/-1?
It's not immediately obvious when reviewing the code that the index the user is thinking about is 1-indexed, whereas the deque is 0-indexed.

Also, are you sure `n_songs_to_remove` is correctly calculated? `+ 1` doesn't make sense to me just looking at the code. Should be caught it tests anyway...

Reply:
Yeah sure 👍 I'll go back, try to make this more explicit, and double check the logic.

The `+1` is just because indexing. What the user sees as the `1st` song in the queue, has index `0` internally.